ops/testing.py

```diff
@@ -2931,23 +2931,49 @@ def add_layer(
         else:
             self._layers[label] = layer_obj
 
-    def _render_services(self) -> Dict[str, pebble.Service]:
-        services: Dict[str, pebble.Service] = {}
+    def _render_services(self) -> Dict[str, pebble.ServiceDict]:
+        services: Dict[str, pebble.ServiceDict] = {}
         for key in sorted(self._layers.keys()):
             layer = self._layers[key]
             for name, service in layer.services.items():
                 # TODO: (jam) 2021-04-07 have a way to merge existing services
-                services[name] = service
+                services[name] = service.to_dict()
```

Comment:
Hmm, not an issue with this PR, but looking at this code made me realize we're not even attempting to combine layers correctly here. We already have the code to combine a layer ([in `_TestingPebbleClient.add_layer`](https://github.com/canonical/operator/blob/dc47640f8ea915049d9e20a1fe4ea189c96ae8d1/ops/testing.py#L2912-L2929)), we just need to use it here as well.

We also have `pebble.Service._merge`, but we'll need similar `_merge` methods on `pebble.Check` and `pebble.LogTarget` and update appropriately to merge those blocks as well.

I think we can leave it "broken" in this PR (it's already broken now, so this doesn't make it worse), but I've opened [#1112](https://github.com/canonical/operator/issues/1112) to track.